app/cdap/components/WrangleHome/services/getWidgetData.tsx

```diff
@@ -0,0 +1,172 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import { getCategorizedConnections } from 'components/Connections/Browser/SidePanel/apiHelpers';
+import {
+  fetchAllConnectorPluginProperties,
+  fetchConnectionDetails,
+  fetchConnectors,
+  getMapOfConnectorToPluginProperties,
+  getSelectedConnectorDisplayName,
+} from 'components/Connections/Create/reducer';
+import WidgetSVG from 'components/WrangleHome/Components/WidgetSVG';
+import { getCategoriesToConnectorsMap } from 'components/WrangleHome/Components/WidgetSVG/utils';
+import { AddConnectionIcon } from 'components/WrangleHome/Components/WrangleCard/iconStore/AddConnectionIcon';
+import { ImportDataIcon } from 'components/WrangleHome/Components/WrangleCard/iconStore/ImportDataIcon';
+import { ImportDatasetIcon } from 'components/WrangleHome/Components/WrangleCard/iconStore/ImportDatasetIcon';
+import {
+  IConnectorArray,
+  IConnectorDetailPayloadArray,
+} from 'components/WrangleHome/Components/WrangleCard/types';
+import React from 'react';
+
+export const getWidgetData = async (cbUpdateState) => {
+  const connectorTypes = await fetchConnectors();
+  const categorizedConnections = await getCategorizedConnections();
+
+  const connectorTypeWithConnections = [];
+  categorizedConnections?.forEach((value, key) => {
+    let mostUpdatedTimeStamp = value[0].updatedTimeMillis;
+    value.forEach((e) => {
+      if (mostUpdatedTimeStamp < e.updatedTimeMillis) {
+        mostUpdatedTimeStamp = e.updatedTimeMillis;
+      }
+    });
+    connectorTypeWithConnections.push({ name: key, time: mostUpdatedTimeStamp });
+  });
+  const sortedConections = connectorTypeWithConnections.slice(0);
+  sortedConections.sort((a, b) => b.time - a.time);
+
+  const connectorDataArray = [];
+  let connectorDataWithSvgArray: IConnectorArray[] = [];
+  const allConnectorsPluginProperties: Map<
+    string,
+    IConnectorDetailPayloadArray[]
+  > = getCategoriesToConnectorsMap(connectorTypes);
+  const connectionPayloadArray: IConnectorDetailPayloadArray[] = [];
+  allConnectorsPluginProperties?.forEach((connectorsArray) => {
+    if (connectorsArray.length) {
+      connectorsArray.map((item) => {
+        connectionPayloadArray.push(item);
+      });
+    }
+  });
+
+  const connectionDetailsData = await Promise.all(
+    connectionPayloadArray.map(async (item, index) => {
+      const selectedConnector = {
+        artifact: item.artifact,
+        category: item.category,
+        name: item.name,
+        type: item.type,
+      };
+      connectorDataArray.push(selectedConnector);
+      return new Promise((resolve, reject) => {
+        const response = fetchConnectionDetails(selectedConnector);
```

Comment:
you can't just return fetchConnectionDetails here?

Reply:
updated the code